monetio/models/icap_mme.py

```diff
@@ -0,0 +1,189 @@
+import pandas as pd
+
+server = "ftp.star.nesdis.noaa.gov"
+base_dir = "/pub/smcd/VIIRS_Aerosol/npp.viirs.aerosol.data/epsaot550/"
+
+valid_filetypes = ("MMC", "C4", "MME")
+valid_data_vars = (
+    "modeaod550",
+    "dustaod550",
+    "pm",
+    "seasaltaod550",
+    "smokeaod550",
+    "totaldustaod550",
+)
+
+
+def build_urls(dates, *, filetype="MMC", data_var="dustaod550"):
+    """Construct URLs for downloading NEPS data.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex or iterable of datetime
+        Dates to download data for.
+    filetype : {'MMC', 'C4', 'MME'}, optional
+    data_var : {'modeaod550', 'dustaod550', 'pm', 'seasaltaod550', \
+        'smokeaod550', 'totaldustaod550'}, optional
+
+    Returns
+    -------
+    urls : pd.Series
+    fnames : pd.Series
+    """
+
+    from collections.abc import Iterable
+
+    if isinstance(dates, Iterable):
+        dates = pd.DatetimeIndex(dates)
+    else:
+        dates = pd.DatetimeIndex([dates])
+
+    urls = []
+    fnames = []
+    print("Building ICAP-MME URLs...")
+    base_url = "https://usgodae.org/ftp/outgoing/nrl/ICAP-MME/"
+
+    for dt in dates:
+        fname = "icap_{}_{}_{}.nc".format(
+            dt.strftime(r"%Y%m%d%H"), filetype.upper(), data_var.lower()
+        )
+        url = base_url + dt.strftime(r"%Y/%Y%m/") + fname
+        urls.append(url)
+        fnames.append(fname)
+
+    # Note: files needed for comparison
+    urls = pd.Series(urls, index=None)
+    fnames = pd.Series(fnames, index=None)
+    return urls, fnames
+
+
+def check_remote_file_exists(file_url):
+    import requests
+
+    r = requests.head(file_url, stream=True, verify=False)
+
+    if r.status_code == 200:
+        return True
+    else:
+        print(f"HTTP Error {r.status_code} - {r.reason}")
+        return False
+
+
+def retrieve(url, fname):
+    """Download the file at `url` to path `fname` if it doesn't exist.
+
+    Parameters
+    ----------
+    url : str
+    fname : str or path-like
+
+    Returns
+    -------
+    None
+    """
+    import os
+
+    import requests
+
+    if not os.path.isfile(fname):
+        print("\n Retrieving: " + fname)
+        print(url)
+        print("\n")
+        r = requests.get(url)
+        r.raise_for_status()
+        with open(fname, "wb") as f:
+            f.write(r.content)
+    else:
+        print("\n File Exists: " + fname)
+
+
+def open_dataset(date, product="MMC", data_var="modeaod550"):
+    """
+    Parameters
+    ----------
+    date : str or datetime-like
+        The date for which to open the dataset.
+        2022-10-29 to current is available.
```

Comment:
Looks like the first non-MME is 2022-09-01
and the first MME is 2014-11-28

Reply:
Also we should provide a description of what MME, C4, MMC mean

Reply:
MME is the same thing as MMC but for whatever reason the changed the name from MultiModelEnsemble to MultiModelConcensus

Reply:
C4 is the Consensus from the 4 models with DA

Reply:
I added the date info to the docstrings.